From 37cb23dba61049b20ce959795432d93fac42931b Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 23 Mar 2021 11:11:30 +0200
Subject: [PATCH] [SW-36711] greco: change load address of uboot

Currently, fit image is copied at SRAM offset 0x200000 and uboot
is loaded at SRAM offset 0x100000. This works as expected till
the size of uboot is less than 1MB. In the cases where uboot size
increases beyond 1MB, uboot overwrites FIT contents stored at
offset 0x200000 which leads to preboot unable to find images of
zephyr.

In order to keep fit image load location same as it is common across all
devices, this patch changes uboot load offset to 0x400000 for greco
onwards devices.

Change-Id: Ie1451e492f46acdc7e930160adbd4c711fe5e890
---

diff --git a/board/habanalabs/common/uboot-zephyr.its b/board/habanalabs/common/uboot-zephyr.its
index bd17d4a..a4ca800 100644
--- a/board/habanalabs/common/uboot-zephyr.its
+++ b/board/habanalabs/common/uboot-zephyr.its
@@ -101,7 +101,7 @@
 			type = "standalone";
 			arch = "arm64";
 			compression = "none";
-			loadaddr = <0x7F 0xF0100000>;
+			loadaddr = <0x7F 0xF0400000>;
 			hash {
 				algo = "sha256";
 			};
diff --git a/board/habanalabs/greco/uboot_asic.fragment b/board/habanalabs/greco/uboot_asic.fragment
index 3f177b5..f41fad0 100644
--- a/board/habanalabs/greco/uboot_asic.fragment
+++ b/board/habanalabs/greco/uboot_asic.fragment
@@ -1,4 +1,5 @@
 CONFIG_SYS_CLK_FREQ=50000000
+CONFIG_SYS_TEXT_BASE=0x7ff0400000
 CONFIG_CONARGS="serial0:115200n8"
 CONFIG_BAUDRATE=115200
 CONFIG_BOOTDELAY=0
